File: level2/JadenCase_문자열_만들기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(string s) {
    string answer = "";
    string word = "";
    vector<string> words;

    for(auto x : s){
        if(x == ' '){
            words.push_back(word);
            word = "";
        }
        else{
            if(word.size() == 0){
                if(isdigit(x) || isupper(x))
                    word = x;
                else
                    word = toupper(x);
            }
            else{
                if(islower(x))
                    word += x;
                else
                    word += tolower(x);
            }
        }
    }

    words.push_back(word);
    reverse(words.begin(), words.end());

    while(!words.empty()){
        answer += words.back();
        answer += " ";
        words.pop_back();
    }
    return answer.substr (0, answer.size()-1);
}

File: JavaScript/level2/JadenCase_문자열_만들기.js
function solution(s) {
    var answer = '';
    
    for(let i=0; i<s.length; i++){
        if(i === 0 && ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z')))
            answer += s[i].toUpperCase();
        else if(i !== 0 && s[i-1] === " " && ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z')))
            answer += s[i].toUpperCase();
        else if(s[i] >= 'A' && s[i] <= 'Z')
            answer += s[i].toLowerCase();
        else
            answer += s[i];   
    }
 
    return answer;
}
